### ISSUE TITLE
Please add support for ghc-9.10.x

### ISSUE THREAD

**Issue by peti** [simonmichael/hledger#2291]
It appears that there is no release of hledger that can be built with ghc-9.10.1. I've tried bumping the version constraints and this works for hledger-lib and hledger, but hledger-ui won't compile:

~~~
[   32s] Hledger/UI/UIUtils.hs:382:28: error: [GHC-87543]
[   32s]     Ambiguous occurrence ‘List’.
[   32s]     It could refer to
[   32s]        either ‘Brick.Widgets.List.List’,
[   32s]               imported from ‘Brick.Widgets.List’ at Hledger/UI/UIUtils.hs:53:28-31,
[   32s]            or ‘Data.List.List’,
[   32s]               imported from ‘Data.List’ at Hledger/UI/UIUtils.hs:56:1-16
[   32s]               (and originally defined in ‘ghc-prim-0.11.0:GHC.Types’).
[   32s]     |
[   32s] 382 | scrollSelectionToMiddle :: List Name item -> EventM Name UIState ()
[   32s]     |                            ^^^^
[   32s] 
[   32s] Hledger/UI/UIUtils.hs:432:66: error: [GHC-87543]
[   32s]     Ambiguous occurrence ‘List’.
[   32s]     It could refer to
[   32s]        either ‘Brick.Widgets.List.List’,
[   32s]               imported from ‘Brick.Widgets.List’ at Hledger/UI/UIUtils.hs:53:28-31,
[   32s]            or ‘Data.List.List’,
[   32s]               imported from ‘Data.List’ at Hledger/UI/UIUtils.hs:56:1-16
[   32s]               (and originally defined in ‘ghc-prim-0.11.0:GHC.Types’).
[   32s]     |
[   32s] 432 | listScrollPushingSelection :: Name -> Int -> Int -> EventM Name (List Name item) (GenericList Name Vector item)
[   32s]     |                                                                  ^^^^
[   32s] 
~~~

Would it be possible to get a new release that fixes this issue?

**Comment by simonmichael** [comment]
Sorry about that, I have been avoiding the work of a minor release; I plan to release it in the dec 1-ish hledger-1.41 release.

**Comment by simonmichael** [comment]
This is resolved by today's 1.41 release.
